x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -5,19 +5,105 @@
  * 2.0.
  */
 
-import React, { memo } from 'react';
-import { EuiSpacer } from '@elastic/eui';
+import React, { memo, useMemo, useState } from 'react';
+import type { EuiButtonGroupOptionProps } from '@elastic/eui';
+import { EuiButtonGroup, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FlyoutPanelProps, PanelPath } from '@kbn/expandable-flyout';
+import { useExpandableFlyoutState } from '@kbn/expandable-flyout';
+import { i18n } from '@kbn/i18n';
+import type { FlyoutPanels } from '@kbn/expandable-flyout/src/store/state';
 import { MisconfigurationFindingsDetailsTable } from './misconfiguration_findings_details_table';
+import { VulnerabilitiesFindingsDetailsTable } from './vulnerabilities_findings_details_table';
 
 /**
  * Insights view displayed in the document details expandable flyout left section
  */
+
+interface CspFlyoutPanelProps extends FlyoutPanelProps {
+  params: {
+    path: PanelPath;
+    hasMisconfigurationFindings: boolean;
+    hasVulnerabilitiesFindings: boolean;
+  };
+}
+interface CspFlyoutPanels extends FlyoutPanels {
+  left: CspFlyoutPanelProps;
+}
+
 export const InsightsTabCsp = memo(
   ({ name, fieldName }: { name: string; fieldName: 'host.name' | 'user.name' }) => {
+    const panels = useExpandableFlyoutState() as CspFlyoutPanels;
+    const defaultTab = () =>
+      panels.left?.params?.hasMisconfigurationFindings
+        ? 'misconfigurationTabId'
+        : panels.left?.params?.hasVulnerabilitiesFindings
+        ? 'vulnerabilitiesTabId'
+        : 'undefined';
+
+    const [activeInsightsId, setActiveInsightsId] = useState(
+      panels.left?.params?.path?.subTab || defaultTab
```

Comment:
im not sure if this works since `defaultTab` is a function definition. you need to be executing it to actually get a value here